addons/xm157/functions/fnc_ballistics_calculator.sqf

```diff
@@ -0,0 +1,99 @@
+#include "script_component.hpp"
+/*
+ * Author: PabstMirror, Ruthberg (Based on ace_atragmx_fnc_calculate_solution)
+ * Calculates elevation and windage
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call ace_xm157_fnc_ballistics_calculator
+ *
+ * Public: No
+ */
+
+
+params ["_targetRange", "_directionOfFire", "_inclinationAngle", "_bank"];
+
+private _weaponInfo = [] call FUNC(ballistics_getData);
+if (_weaponInfo isEqualTo []) exitWith { [0,0] };
+_weaponInfo params ["_scopeBaseAngle","_boreHeight","_airFriction","_muzzleVelocity","_bc",
+    "_dragModel","_atmosphereModel","_barrelTwist","_twistDirection","_caliber","_bulletLength","_bulletMass"];
+
+private _latitude = GVAR(data) getOrDefault ["latitude", 0];
+
+// Get Wind
+private _windSpeed = GVAR(data) getOrDefault ["wind_speed", 0];
+private _windDirection = 22.5 * (GVAR(data) getOrDefault ["wind_dir", 0]);
+private _wind = [sin (_directionOfFire-_windDirection) * _windSpeed, -cos (_directionOfFire-_windDirection) * _windSpeed, 0];
+
+// Get atmosphere
+private _altitude = (getPosASL ace_player) select 2;
+private _relativeHumidity = EGVAR(weather,currentHumidity);
+private _temperature = _altitude call EFUNC(weather,calculateTemperatureAtHeight);
+private _barometricPressure = _altitude call EFUNC(weather,calculateBarometricPressure);
+
+
+private _bulletPos = [0,0,-(_boreHeight / 100)];
+private _lastBulletPos = +_bulletPos;
+private _bulletVelocity = [0,Cos(_scopeBaseAngle) * _muzzleVelocity,Sin(_scopeBaseAngle) * _muzzleVelocity];
+private _gravity = [-sin (_bank) * cos(_scopeBaseAngle + _inclinationAngle) * -GRAVITY, 
+                    sin(_scopeBaseAngle + _inclinationAngle) * -GRAVITY, 
+                    cos (_bank) * cos(_scopeBaseAngle + _inclinationAngle) * -GRAVITY];
```

Comment:
```suggestion
private _angleGravity = (_scopeBaseAngle + _inclinationAngle) * -GRAVITY;
private _gravity = [-sin (_bank) * cos (_angleGravity), 
                    sin (_angleGravity), 
                    cos (_bank) * cos (_angleGravity)];
```
(`_angleGravity` might be an unimaginative or even a misleading name, though.)

Reply:
don't want `GRAVITY` in that term

Reply:
this func base very closely on https://github.com/acemod/ACE3/blob/master/addons/atragmx/functions/fnc_calculate_solution.sqf#L61

tbh I'm not really sure why that modifies gravity instead of initial velocity, but I trust ulteq's work so I'd like to keep this func as close to that as possible